apps/aecore/src/aec_conductor.erl

```diff
@@ -625,9 +659,91 @@ retry_mining_with_new_nonce(Candidate, State) ->
     NewCandidate = Candidate#candidate{nonce = NewNonce},
     start_mining(State#state{block_candidate = NewCandidate}).
 
+%%%===================================================================
+%%% Worker: Start signing microblocks
+
+start_micro_signing(#state{keys_ready = false} = State) ->
+    %% We need to get the keys first
+    wait_for_keys(State);
+start_micro_signing(#state{mining_state = 'stopped'} = State) ->
+    State;
+start_micro_signing(#state{consensus = #consensus{leader = true}, micro_block_candidate = undefined} = State) ->
+    %% We need to generate a new block candidate first.
+    create_micro_block_candidate(State);
+start_micro_signing(#state{consensus = #consensus{leader = true},
+                           block_candidate = #candidate{top_hash = OldHash},
+                           seen_top_block_hash = SeenHash
+                          } = State) when OldHash =/= SeenHash ->
+    %% Probably no longer the leader
+    epoch_mining:debug("Candidate is not refering to previous hash"),
+    create_micro_block_candidate(State#state{block_candidate = undefined});
+start_micro_signing(#state{micro_block_candidate = Candidate} = State) ->
+    epoch_mining:info("Starting signing round"),
+    HeaderBin = Candidate#candidate.bin,
+    Info      = [{top_block_hash, State#state.seen_top_block_hash}],
+    aec_events:publish(start_micro_signing, Info),
+    Fun = fun() ->
+                  {aec_keys:sign(HeaderBin), HeaderBin}
+          end,
+    dispatch_worker(micro_signing, Fun, State).
+
+handle_micro_signing_reply(_Reply, #state{micro_block_candidate = undefined} = State) ->
+    %% Something invalidated the block candidate already.
+    start_micro_sleep(State);
+handle_micro_signing_reply({ok, {[Signature], HeaderBin}}, #state{} = State) ->
```

Comment:
`aec_keys:sign(HeaderBin)` returns `{ok, {HeaderBin, Singatures}}` so this should be changed to:
```
handle_micro_signing_reply({ok, {HeaderBin, [Signature]}}, #state{} = State)
```